src/lockfile/wrapper.js

```diff
@@ -156,7 +160,8 @@ export default class Lockfile {
       invariant(ref, 'Package is missing a reference');
       invariant(remote, 'Package is missing a remote');
 
-      const seenPattern = remote.resolved && seen.get(remote.resolved);
+      const altRemoteKey = keyForRemote(remote);
```

Comment:
I think we should merge all this functionality into `keyForRemote` and just use it everywhere without the `remote.resolved` checks. What do you think?